src/service/resources-impl.js

```diff
@@ -387,15 +391,42 @@ export class Resources {
     this.resources_.push(resource);
 
     if (this.isRuntimeOn_) {
-      // Try to immediately build element, it may already be ready.
-      resource.build(this.forceBuild_);
-      this.schedulePass();
+      if (this.documentReady_) {
+        // Try to immediately build element, it may already be ready.
+        resource.build(this.forceBuild_);
```

Comment:
Done. Should I go ahead and remove the usages for `isReadyToBuild` across AMP? Or should I do that in a separate PR?